### PR TITLE
add "decoded" option to string input

### PR DESCRIPTION
**Pull request by qbalukom**

### CODE REVIEW

File: src/components/Extensibility/components-form/StringRenderer.js [hunk 1]
```diff
@@ -16,11 +20,31 @@ export function StringRenderer({
...
+  const [decoded] = jsonata(
+    schema.get('decoded'),
+    itemVars(originalResource, schema.get('itemVars'), storeKeys),
+  );
+
+  if (decoded) {
+    try {
+      value = base64Decode(value);
+    } catch (e) {
+      // noop
```
Comment (mrCherry97) [inline src/components/Extensibility/components-form/StringRenderer.js:45]: image

File: docs/extensibility/form-section.md [hunk 2]
```diff
@@ -207,6 +207,7 @@ Text widgets render a field as a text field. They are used by default for all st
 - **inputInfo** - a string below the input field that shows how to fill in the input.
 - **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
 - **readOnly** - a boolean which specifies if a field is read-only. Defaults to `false`.
+- **decoded** - a [JSONata](jsonata.md) expression stating whether the field should be decoded from a base64 value in the resource.
```
Comment (mrCherry97) [inline docs/extensibility/form-section.md:210]: Can we provide example usage to docs?

image

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
image

File: src/components/Extensibility/components-form/StringRenderer.js (hunk 1 above)
Comment (qbalukom) [reply src/components/Extensibility/components-form/StringRenderer.js:45]: we have to catch the error, but we don't want to do anything, thus the noop

**Review by Lyczeq — CHANGES_REQUESTED** [review]

File: docs/extensibility/form-section.md [hunk 3]
```diff
@@ -207,6 +207,8 @@ Text widgets render a field as a text field. They are used by default for all st
 - **inputInfo** - a string below the input field that shows how to fill in the input.
 - **description** - a string displayed in a tooltip when you hover over a question mark icon, next to the input's label. The default value is taken from the CustomResourceDefintion (CRD).
 - **readOnly** - a boolean which specifies if a field is read-only. Defaults to `false`.
+- **decodable** - a boolean which specifies that the field is base64-encoded and can be decoded in UI. Can't be used together with **enum**.
```
[the remark this answers is not in the document]
Comment (qbalukom) [reply docs/extensibility/form-section.md]: we don't always state what it defaults to. I'd say it's rather implied to be false, and a bit redundant to state that... but that's just my opinion

[the remark this answers is not in the document]
Comment (Lyczeq) [reply docs/extensibility/form-section.md]: That's right. Let's wait for the TW opinion

**Review by Lyczeq — DISMISSED** [review]

**Review by Lyczeq — DISMISSED** [review]

**Review by grego952 — APPROVED** [review]

**Review by Lyczeq — APPROVED** [review]

**Pull request merged**